web/modules/custom/dept_migrate/modules/dept_migrate_group_files/src/EventSubscriber/MediaEntityMigrateSubscriber.php

```diff
@@ -0,0 +1,82 @@
+<?php
+
+namespace Drupal\dept_migrate_group_files\EventSubscriber;
+
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+use Drupal\Core\Logger\LoggerChannelFactory;
+use Drupal\dept_migrate\MigrateSupport;
+use Drupal\media\MediaInterface;
+use Drupal\migrate\Event\MigrateEvents;
+use Drupal\migrate\Event\MigratePostRowSaveEvent;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+
+/**
+ * Class MediaEntityMigrateSubscriber.
+ */
+class MediaEntityMigrateSubscriber implements EventSubscriberInterface {
+
+  /**
+   * @var \Drupal\Core\Entity\EntityTypeManagerInterface
+   */
+  protected $entityTypeManager;
+
+  /**
+   * @var \Drupal\dept_migrate\MigrateSupport
+   */
+  protected $migrateSupport;
+
+  /**
+   * Drupal\Core\Logger\LoggerChannelFactory definition.
+   *
+   * @var \Drupal\Core\Logger\LoggerChannelFactory
+   */
+  protected $logger;
+
+  /**
+   * @param \Drupal\Core\Entity\EntityTypeManagerInterface $entity_type_manager
+   *   Entity type manager service.
+   * @param \Drupal\dept_migrate\MigrateSupport $migrate_support
+   *   Migrate support service.
+   * @param \Drupal\Core\Logger\LoggerChannelFactory $logger
+   *   Drupal logger.
+   */
+  public function __construct(EntityTypeManagerInterface $entity_type_manager,
+                              MigrateSupport $migrate_support,
+                              LoggerChannelFactory $logger) {
+
+    $this->entityTypeManager = $entity_type_manager;
+    $this->migrateSupport = $migrate_support;
+    $this->logger = $logger->get('dept_migrate');
+  }
+
+  /**
+   * Callback for post-row-save event.
+   *
+   * @param \Drupal\migrate\Event\MigratePostRowSaveEvent $event
+   *   The migrate post row save event.
+   */
+  public function onPostRowSave(MigratePostRowSaveEvent $event) {
+    if (preg_match('/^d7_file_media_/', $event->getMigration()->id())) {
```

Comment:
Rather than use dedicated migrate configs to create the group content/relation entities, this event subscriber adds them for any media migration. Benefits: more control + visibility, and cleaner migrate:status output.